WalletWasabi/WabiSabi/Crypto/CredentialIssuer.cs

```diff
@@ -58,7 +60,11 @@ public class CredentialIssuer
 			MaxAmount = maxAmount;
 			RangeProofWidth = (int)Math.Ceiling(Math.Log2(MaxAmount));
 			CredentialIssuerSecretKey = Guard.NotNull(nameof(credentialIssuerSecretKey), credentialIssuerSecretKey);
-			NumberOfCredentials = Guard.InRangeAndNotNull(nameof(numberOfCredentials), numberOfCredentials, 1, 100);
+			if (numberOfCredentials != CredentialsPerRequest)
```

Comment:
the `numberOfCredentials` parameter could just be removed from the `CredentialIssuer` constructor if we're fixing it to a constant. however, that would force us to remove the tests for the case where the case of `k=3` is exercised. we could add a comment so that git blame will easily show where to find them, as we might find a reason to change it to 3 later (i'm skeptical at this point though).

if we really want to keep the tests we need to keep the parameter and just enforce that the value is equal to the constant at the arena level instead of here.

Reply:
NACK. The `CredentialIssuer` works okay and there is no reason to hardcode it a constant for a value that it receives as a parameter.

Reply:
i don't think that change is actually on the table.

we have two possibilities:

- remove the parameter and only use a constant, with the reasoning that we just say k=2 and deleting code is a good thing
- keep the parameter, and don't add a constant, with the reasoning that k=3 is still a possibility and we don't want to discard those tests

having both a constant and a parameter doesn't make sense to me, it only adds complexity

Reply:
In those cases where we want a parameter to be always the same for our implementation we don't remove the parameter, we simplify the interface by using some default value instead. There are many ways to do that, we could create a factory method that creates a new instance with `k=2`, or a new constructor, etc. 

Moreover, we inject the random instance because it is useful only for testing so, in production we should never inject other than `SecureRandom`, right? so, why not create a factory like:

```c#
/// CredentialIssuer.cs file

public CredentialIssuer Create(CredentialIssuerSecretKey credentialIssuerSecretKey, ulong maxAmount)
{
    return new CredentialIssuer(
        credentialIssuerSecretKey,: credentialIssuerSecretKey, 
        numberOfCredentials: 2,
	randomNumberGenerator: SecureRandom.Random
	maxAmount: maxAmount);
}

```

Reply:
I saw your answer now. I added a default parameter and set the default to the constant.

Reply:
Perhaps I put too much emphasis in something like this, after all it is just a detail. The point is that the ability to use different values for `k` has been very useful for playing with the library, benchmark it and also test it with different `k` values, that's why I don't want to limit it. We know that an algorithm that works okay with any number works well with a specific value but the other way around is not necessarily true. Also, if well we agreed that `k=2` is okay, it could be possible or even necessary to use a different value in case we find an algorithm that works better with values other than 2 or in case the network roundtrips are too much expensive or who knows what other reason.

Btw, if tomorrow I need to run my benchmark project again to verify something then I could need to change the code back again to make the class work with any `k`. This is not a problem but it sucks anyway.

Reply:
ack, this seems stronger than my reasons for removing the parameter, unless @molnard disagrees i am happy with keeping it as a parameter